### ISSUE TITLE
Template render has incorrect bootstrap flag when upgrading with one replica

### ISSUE THREAD

**Issue by mook-as** [cloudfoundry-incubator/quarks-operator#1177]
**Describe the bug**
When upgrading a HA KubeCF, the blobstore pre-start does not run correctly, as it only has one instance, and that one instance does not act like it is the bootstrap instance.

**To Reproduce**
- Deploy KubeCF v2.5.3 (HA on AKS)
- Upgrade to operator v6.1.14
- Upgrade to KubeCF v2.5.6

Relevant environment that may lead to hints:
Environment variable | value
--- | ---
`KUBE_AZ` | (unset)
`BOSH_AZ` | (unset)
`CF_OPERATOR_AZ` | (unset)
`INITIAL_ROLLOUT` | `false`
`REPLICAS` | `1`
`AZ_INDEX` | `1`

**Expected behavior**
In [blobstore pre-start](https://github.com/cloudfoundry/capi-release/blob/1.99.0/jobs/blobstore/templates/pre-start.sh.erb#L33), given that there is only one node, it should run `setup_blobstore_directories`; instead, it prints `Blobstore job is only run on the bootstrap instance`.

**Environment**
 - CF-Operator Version v6.1.14
 - KubeCF 2.5.6

**Additional context**
Possible call stack (have not confirmed this is the issue):
- [`cf-operator util template-render`](https://github.com/cloudfoundry-incubator/quarks-operator/blob/v6.1.14/cmd/internal/trender.go#L104)
- [`manifest.RenderJobTemplates()`](https://github.com/cloudfoundry-incubator/quarks-operator/blob/v6.1.14/pkg/bosh/manifest/cmd_render_job_tmpls.go#L71)
- [`manifest.InstanceGroup.jobInstances()`](https://github.com/cloudfoundry-incubator/quarks-operator/blob/v6.1.14/pkg/bosh/manifest/instance_group.go#L115)
has:

https://github.com/cloudfoundry-incubator/quarks-operator/blob/v6.1.14/pkg/bosh/manifest/instance_group.go#L121-L132

```go

	bootstrapIndex := 0
	if !initialRollout {
		bootstrapIndex = ig.Instances*len(ig.AZs) - 1
	}

	if len(ig.AZs) > 0 {
		for azIndex, az := range ig.AZs {
			jobsInstances = ig.generateJobInstances(jobsInstances, azIndex, az, jobName, bootstrapIndex)
		}
	} else {
		jobsInstances = ig.generateJobInstances(jobsInstances, -1, "", jobName, bootstrapIndex)
	}
```
In the `ig.AZs == 0` case (which must be a thing, since there's a branch for it), when `initialRollout` is `false`, `bootstrapIndex` is `1 * 0 - 1` → `-1`, which is never true for any replica.

**Comment by mook-as** [comment]
Fixed in 89a56300d40a0da74719c9d80c1e7e27616fc68a; leaving open because it's not been merged to master (and hasn't got a test).

**Comment by manno** [comment]
Merged into master now: https://github.com/cloudfoundry-incubator/quarks-operator/pull/1222
